src/services/crawl-dids/dids.service.ts

```diff
@@ -38,25 +30,44 @@ export default class DidDatabaseService extends BullableService {
             throw new Error('DID is required for deletion');
         }
 
-        this.logger.info(`Attempting to delete DID: ${ctx.params}`);
+        this.logger.info(`Attempting to delete DID: ${did}`);
 
-        const deletedRows = await knex('dids').where({ did }).del();
+        const deletedRows = await knex('dids')
+            .where({ did })
+            .update({ is_deleted: true, deleted_at: new Date().toISOString(), event_type: "remove_did" });
 
         if (deletedRows === 0) {
             this.logger.warn(`No DID found to delete: ${did}`);
             return { success: false, message: `No record found for DID: ${did}` };
         }
 
-        this.logger.info(`Deleted DID: ${did}`);
+        this.logger.info(`Marked DID as deleted: ${did}`);
         return { success: true };
     }
 
-
-    @Action({
-        name: 'get'
-    })
+    @Action({ name: 'get' })
     async getDid(ctx: any) {
         const { did } = ctx.params;
         return await knex('dids').where({ did }).first();
     }
-}
\ No newline at end of file
+
+
+    @Action({ rest: "GET get/:did", params: { did: "string" } })
+    async getSingleDid(ctx: Context<{ did: string }>) {
+        const did = await knex("dids").where({ did: ctx.params.did }).first();
+        if (!did) throw new Error("DID not found");
```

Comment:
Here are you throwing a 404 or a 500? I don't think we should send any contents into response body in case of error: just the code (unless the spec says otherwise)